linkerd/system/src/linux.rs

```diff
@@ -55,3 +58,13 @@ fn sysconf(num: libc::c_int, name: &'static str) -> Result<u64, io::Error> {
         val => Ok(val as u64),
     }
 }
+
+fn or(soft_limit: process::LimitValue, hard_limit: process::LimitValue) -> Option<u64> {
```

Comment:
Looks like we are always returning `Some` inside of the `match`. We could just wrap the entire thing in an option, or just do `Ok(Some(max_fds))` on line 47.